python/SNS_ENCRYPTED_TOPIC_CHECK/SNS_ENCRYPTED_TOPIC_CHECK.py

```diff
@@ -0,0 +1,451 @@
+
+Scenarios:
+  Scenario: 1
+     Given: No SNS topic is present
+      Then: Return NOT_APPLICABLE
+  Scenario: 2
+     Given: KmsKeyId parameter is configured
+       And: KmsKeyId does not contain valid KMS Key ARN(s)
+      Then: Return ERROR
+  Scenario: 3
+     Given: There is at least 1 SNS topic
+       And: SNS topic is not encrypted
+      Then: Return NON_COMPLIANT with Annotation "The Amazon Simple Notification Service topic is not encrypted."
+  Scenario: 4
+     Given: There is at least 1 SNS topic
+       And: SNS topic is encrypted
+       And: KmsKeyId parameter is not configured
+      Then: Return COMPLIANT
+  Scenario: 5
+     Given: There is at least 1 SNS topic
+       And: SNS topic is encrypted
+       And: KmsKeyId parameter is configured
+       And: SNS topic encryption Key not matching any KMS key(s) in KmsKeyId
+      Then: Return NON_COMPLIANT with Annotation "This SNS topic is not encrypted with KMS Key {KmsKeyId}."
+  Scenario: 6
+     Given: There is at least 1 SNS topic
+       And: SNS topic is encrypted
+       And: KmsKeyId parameter is configured
+       And: SNS topic encryption Key matching one of the KMS key in KmsKeyId
+      Then: Return COMPLIANT
+'''
+
+import json
+import sys
+import datetime
+import boto3
+import botocore
+
+try:
+    import liblogging
+except ImportError:
+    pass
+
+##############
+# Parameters #
+##############
+
+# Define the default resource to report to Config Rules
+DEFAULT_RESOURCE_TYPE = 'AWS::SNS::Topic'
+
+# Set to True to get the lambda to assume the Role attached on the Config Service (useful for cross-account).
+ASSUME_ROLE_MODE = False
+
+# Other parameters (no change needed)
+CONFIG_ROLE_TIMEOUT_SECONDS = 900
+
+#############
+# Main Code #
+#############
+
+def evaluate_compliance(event, configuration_item, valid_rule_parameters):
+    sns_client = get_client('sns', event)
+    topic_list_of_dict = get_all_topic(sns_client)
+    result = []
+    if not topic_list_of_dict:
+        return build_evaluation(
+            event['accountId'],
+            'NOT_APPLICABLE',
+            event,
+            resource_type='AWS::::Account'
+        )
+
+    for topic_dict in topic_list_of_dict:
+        response_topic_attributes_dict = sns_client.get_topic_attributes(TopicArn=topic_dict['TopicArn'])
+
+        if "KmsMasterKeyId" not in response_topic_attributes_dict['Attributes']:
+            result.append(build_evaluation(
+                topic_dict['TopicArn'],
+                'NON_COMPLIANT',
+                event,
+                annotation="The Amazon Simple Notification Service topic is not encrypted."
+            ))
+            continue
+
+        if not valid_rule_parameters:
+            result.append(build_evaluation(topic_dict['TopicArn'], 'COMPLIANT', event))
+            continue
+
+        if "KmsMasterKeyId" in response_topic_attributes_dict['Attributes'] and \

```

Comment:
You don't need this `"KmsMasterKeyId" in response_topic_attributes_dict['Attributes']`. It is already a given from L102

Reply:
absolutely right ! updated the code